TESTS/netsocket/tls/main.cpp

```diff
@@ -66,7 +67,17 @@ static void _ifup()
     NetworkInterface *net = NetworkInterface::get_default_instance();
     nsapi_error_t err = net->connect();
     TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, err);
-    printf("MBED: TLSClient IP address is '%s'\n", net->get_ip_address() ? net->get_ip_address() : "null");
+    const char *address = net->get_ip_address();
+
+    printf("MBED: TLSClient IP address is '%s'\n", address ? address : "null");
+#if MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == MESH
+    uint8_t addr_hex[16];
+    stoip6(address, strlen(address), addr_hex);
+    if ((addr_hex[0] & 0xe0) != 0x20) {
+        printf("Address is not global  Please check Border Router  TEST SKIP \n");
+        GREENTEA_TESTSUITE_RESULT(false);
```

Comment:
Should this be `TEST_SKIP_MESSAGE("Mesh not working")`

I don't really know the difference why

Reply:
I don't think we want to skip the tests? We just want them to fail with a reasonable message, pointing to the border router issue, to avoid unnecessary debugging/bug reports.